contracts/src/everlong/EverlongERC4626.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+// import { IEverlongERC4626 } from "../interfaces/IEverlongERC4626.sol";
+import { ERC4626 } from "solady/tokens/ERC4626.sol";
+
+contract EverlongERC4626 is ERC4626 {
+    /// @notice Whether virtual shares will be used to mitigate the inflation attack.
+    bool public constant useVirtualShares = true;
+
+    /// @notice Used to reduce the feasibility of an inflation attack.
+    uint8 public constant decimalsOffset = 3;
+
+    /// @notice Address of the underlying Hyperdrive instance.
+    address internal immutable _underlying;
+
+    /// @notice Decimals used by the underlying Hyperdrive asset.
+    uint8 internal immutable _decimals;
+
+    /// @notice Name of the Everlong token.
+    string internal _name;
+
+    /// @notice Symbol of the Everlong token.
+    string internal _symbol;
+
+    constructor(
+        address underlying_,
+        string memory name_,
+        string memory symbol_
+    ) {
+        _underlying = underlying_;
+
+        (bool success, uint8 result) = _tryGetAssetDecimals(underlying_);
+        _decimals = success ? result : _DEFAULT_UNDERLYING_DECIMALS;
+
+        _name = name_;
+        _symbol = symbol_;
```

Comment:
We have three code blocks and zero comments here. Let's make sure to preserve our style.